public/app/plugins/datasource/graphite/components/GraphiteTextEditor.tsx

```diff
@@ -19,6 +19,13 @@ export function GraphiteTextEditor({ rawQuery }: Props) {
     [dispatch]
   );
 
+  const pasteQuery = useCallback(async () => {
+    const query = await navigator.clipboard.readText();
+
+    dispatch(actions.updateQuery({ query }));
+    dispatch(actions.runQuery());
+  }, [dispatch]);
```

Comment:
Not sure if this is the implementation you want. If a user pasts in a partial query, this will completely wipe out the rest of the query, no?

i don't know graphite syntax, but consider they had in the query `SELECT time, event FROM` and needed to paste in the table name - the whole query would be replaced with _just_ the table name.